gwpy/timeseries/timeseries.py

```diff
@@ -1078,11 +1078,13 @@ def filter(self, *filt, **kwargs):
         filtfilt = kwargs.pop('filtfilt', False)
 
         # parse filter
-        form, filt = filter_design.parse_filter(
-            filt,
-            analog=kwargs.pop('analog', False),
-            sample_rate=self.sample_rate.to('Hz').value,
-        )
+        form, filt = filter_design.parse_filter(filt)
+
+        if kwargs.pop('analog', False):
+            form, filt = filter_design.convert_to_digital(
+                form, filt, self.sample_rate.to('Hz').value
+            )
```

Comment:
Please reformat this to one argument per-line, to minimise future diffs.
```suggestion
            form, filt = filter_design.convert_to_digital(
                form,
                filt,
                self.sample_rate.to('Hz').value,
            )
```